components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/artifact/SAMLSSOArtifactResolutionService.java

```diff
@@ -116,26 +132,55 @@ public ArtifactResponse sendArtifactResolveRequest(ArtifactResolve artifactResol
 
         String artifactResponseString = soapMessageService.sendSOAP(envelopeElement, SSOUtils.getArtifactResolveUrl
                 (authenticatorProperties));
-        Pattern pattern = Pattern.compile("<samlp:ArtifactResponse.+</samlp:ArtifactResponse>", Pattern.DOTALL);
-        Matcher matcher = pattern.matcher(artifactResponseString);
-        if (matcher.find()) {
-            String artifactResponseReceived = matcher.group(0);
-            if (log.isDebugEnabled()) {
-                log.debug("Artifact Response: " + artifactResponseReceived);
-            }
-            try {
-                ArtifactResponse artifactResponse = (ArtifactResponse) SSOUtils.unmarshall(artifactResponseReceived);
-                if (isArtifactResponseValid(artifactResolve, artifactResponse)) {
-                    return artifactResponse;
+        ArtifactResponse artifactResponse = extractArtifactResponse(artifactResponseString);
+
+        if (artifactResponse != null && isArtifactResponseValid(artifactResolve, artifactResponse)) {
```

Comment:
Null check is already added inside isArtifactResponseValid() [1], no need to add it in here.

[1] https://github.com/vihanga-liyanage/identity-outbound-auth-samlsso/blob/2cf2a4fe6423a3c1aa3348a8401f9100c3bfb2ca/components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/artifact/SAMLSSOArtifactResolutionService.java#L189